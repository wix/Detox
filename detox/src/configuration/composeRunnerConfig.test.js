// @ts-nocheck
describe('composeRunnerConfig', () => {
  let composeRunnerConfig;
  let cliConfig, globalConfig;

  beforeEach(() => {
    cliConfig = {};
    globalConfig = {};

    composeRunnerConfig = () => require('./composeRunnerConfig')({
      cliConfig,
      globalConfig,
    });
  });

  it('should take .testRunner from globalConfig', () => {
    globalConfig.testRunner = 'jest';
    expect(composeRunnerConfig().testRunner).toBe('jest');
  });

  it('should take .test-runner from globalConfig', () => {
    globalConfig['test-runner'] = 'jest';
    expect(composeRunnerConfig().testRunner).toBe('jest');
  });

  it('should set .testRunner to "mocha" by default', () => {
    expect(composeRunnerConfig().testRunner).toBe('mocha');
  });

  it('should take .runnerConfig from CLI', () => {
    globalConfig.runnerConfig = 'from/config.json';
    cliConfig.runnerConfig = 'from/cli.json';

    expect(composeRunnerConfig().runnerConfig).toBe('from/cli.json');
  });

  it('should take .runnerConfig from config if it is not defined via CLI', () => {
    globalConfig.runnerConfig = 'from/config.json';
    delete cliConfig.runnerConfig;

    expect(composeRunnerConfig().runnerConfig).toBe('from/config.json');
  });

  it('should take .runner-config from config if it is not defined via CLI', () => {
    globalConfig['runner-config'] = 'from/config.json';
    delete cliConfig.runnerConfig;

    expect(composeRunnerConfig().runnerConfig).toBe('from/config.json');
  });

  it('should set .runnerConfig to e2e/mocha.opts if .testRunner is mocha', () => {
    globalConfig.testRunner = 'mocha';
    expect(composeRunnerConfig().runnerConfig).toBe('e2e/mocha.opts');
  });

  it('should set .runnerConfig to e2e/config.json if .testRunner is jest', () => {
    globalConfig.testRunner = 'jest';
    expect(composeRunnerConfig().runnerConfig).toBe('e2e/config.json');
  });

  it('should take .specs from globalConfig', () => {
    globalConfig.specs = 'e2e/suite1';
    expect(composeRunnerConfig().specs).toBe('e2e/suite1');
  });

  it('should treat undefined .specs as an empty string', () => {
    delete globalConfig.specs;
    expect(composeRunnerConfig().specs).toBe('');
  });

});
