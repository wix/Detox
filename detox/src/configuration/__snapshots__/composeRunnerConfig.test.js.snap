// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`composeRunnerConfig unhappy scenarios deprecated "testRunner" in local config 1`] = `
"testRunner should be an object, not a string

HINT: Check that in your Detox config.

{
  configurations: {
    default: {
      testRunner: 'nyc jest'
    }
  }
}"
`;

exports[`composeRunnerConfig unhappy scenarios non-string and non-object "testRunner" in global config 1`] = `
"testRunner should be an object, not a number

HINT: Check that in your Detox config.

{
  testRunner: 0,
  configurations: [Object]
}"
`;
